### PR TITLE
Fix a few documentation errors including in public APIs

### PR DESCRIPTION
**Pull request by ajkr**
The internal WriteBatch doc wrongly indicated which optypes are followed by varstring. Updated some optypes according to the following code: https://github.com/facebook/rocksdb/blob/76383bea5df1136c95babf5f9f40b24f85e9ad8e/db/write_batch.cc#L418-L429

The `Iterator::Refresh()` + `DeleteRange()` bug was fixed in #9258; removed the warnings.

`GetMergeOperands()` does populate `*number_of_operands` including upon successful return: https://github.com/facebook/rocksdb/blob/76383bea5df1136c95babf5f9f40b24f85e9ad8e/db/db_impl/db_impl.cc#L1917-L1919

### CODE REVIEW

File: db/write_batch.cc [hunk 1]
```diff
@@ -22,12 +22,12 @@
 //    kTypeColumnFamilySingleDeletion varint32 varstring
 //    kTypeColumnFamilyRangeDeletion varint32 varstring varstring
 //    kTypeColumnFamilyMerge varint32 varstring varstring
-//    kTypeBeginPrepareXID varstring
-//    kTypeEndPrepareXID
+//    kTypeBeginPrepareXID
+//    kTypeEndPrepareXID varstring
```
Comment (riversand963) [inline db/write_batch.cc:26]: Could you add an entry for `kTypeCommitXIDAndTimestamp varstring varstring`? Thanks!

File: include/rocksdb/db.h [hunk 2]
```diff
@@ -541,16 +537,20 @@ class DB {
...
   }
 
-  // Returns all the merge operands corresponding to the key. If the
-  // number of merge operands in DB is greater than
-  // merge_operands_options.expected_max_number_of_operands
-  // no merge operands are returned and status is Incomplete. Merge operands
-  // returned are in the order of insertion.
-  // merge_operands- Points to an array of at-least
+  // Assigns all the merge operands corresponding to the key to the
+  // `merge_operands` array. Upon success, the number of merge operands set is
```
Comment (riversand963) [inline include/rocksdb/db.h:541]: Is "number of merge operands set" a typo?

**Review by riversand963 — APPROVED** [review]

File: db/write_batch.cc (hunk 1 above)
Comment (ajkr) [reply db/write_batch.cc:26]: Sure, done.

File: include/rocksdb/db.h (hunk 2 above)
Comment (ajkr) [reply include/rocksdb/db.h]: Rewrote to try to make it clearer.

**Review by ajkr — COMMENTED** [review]
Thanks for the review!

**Pull request closed without merging**